### PR TITLE
Bind to bridge interface

### PR DESCRIPTION
**Pull request by caglar10ur**
Otherwise VCH acts like a forwarding DNS server

### CODE REVIEW

**Comment by dougm** [comment]
lgtm

File: lib/dns/dns.go [hunk 1]
```diff
@@ -519,6 +520,18 @@ func (s *Server) ServeDNS(w mdns.ResponseWriter, r *mdns.Msg) {
...
 // Start starts the DNS server
 func (s *Server) Start() {
+	// Call BindToDevice if IP is empty and Interface is set
+	if s.IP == "" && s.Interface != "" {
+		uf, err := s.udpconn.File()
+		if err != nil {
+			log.Errorf("Getting the fd failed with: %s", err)
+		}
+		// BindToDevice binds the socket associated with fd to device.
+		if err := BindToDevice(int(uf.Fd()), s.Interface); err != nil {
```
Comment (hickeng) [inline lib/dns/dns.go]: could be nil pointer (or safe - depends entirely on impl of Fd() )

File: lib/dns/dns.go [hunk 2]
```diff
@@ -534,6 +547,18 @@ func (s *Server) Start() {
...
 	log.Infof("Ready for queries on udp://%s", s.Addr())
 
+	// Call BindToDevice if IP is empty and Interface is set
+	if s.IP == "" && s.Interface != "" {
+		tf, err := s.tcplisten.File()
+		if err != nil {
+			log.Errorf("Getting the fd failed with: %s", err)
+		}
+		// BindToDevice binds the socket associated with fd to device.
+		if err := BindToDevice(int(tf.Fd()), s.Interface); err != nil {
```
Comment (hickeng) [inline lib/dns/dns.go]: ditto

**Comment by hickeng** [comment]
LGTM once comments are addressed

**Comment by caglar10ur** [comment]
♻️

**Pull request merged**